cmd/juju/application/utils/utils.go

```diff
@@ -98,15 +105,43 @@ func getResources(
 	return resources.AsMap(svcs[0].Resources), nil
 }
 
-func filterResources(
+// getAvailableRepositoryResources gets the current resources for this
+// charm available in the repository.
+func getAvailableRepositoryResources(newCharmID application.CharmID, repositoryResourceLister CharmClient) (map[string]charmresource.Resource, error) {
+	if repositoryResourceLister == nil {
+		// not required for local charms
+		return nil, nil
+	}
+	available, err := repositoryResourceLister.ListCharmResources(newCharmID.URL, newCharmID.Origin)
```

Comment:
Does this account for published resources only? ie does it ignore uploaded resources that have not yet been published?

Reply:
This is what the repository (charmhub) says are current resources revisions for the given channel. The resourceLister queries state to see what the current resources are. Uploaded resources are verified in shouldUpgradeResource.